syntax = "proto3";

package bili;

import "base.proto";

message GuardInfo{
    uint32 guard_level = 1;
    string expire_time = 2;
}

message UserBase2 {
    string uname = 1;
    string face = 2;
    UserBase user_base1 = 5;
    UserBase user_base2 = 6;
}

message OnlineRankUserInfo {
    uint32 uid = 1;
    UserBase2 user_base = 2;
    GuardInfo guard_info = 6;
}

message OnlineRankV3Data {
    uint32 uid = 1;
    string face = 2;
    string score = 3;
    string uname = 4;
    int32 rank = 5;
    optional int32 before_rank = 6;
    OnlineRankUserInfo user_info = 8;

}

message OnlineRankV3 {
    string cmd = 1; // online_rank
    repeated OnlineRankV3Data data = 3;
}
